src/kaspad.rs

```diff
@@ -12,6 +12,7 @@ pub struct Config {
     network: Network,
     data_folder: Option<PathBuf>,
     enable_upnp: bool,
+    archival: bool,
```

Comment:
This field needs to be tagged with `#[serde(default)]`, otherwise the previous version of the config file will fail to load (as the field is missing), resetting the entire config (as currently, it will fall back to a general error handling).

The default value for `bool` is `false`.  You changed the `new()` ctor, which is what is used to create a new config, however, when config is deserialized, it uses serde.

Reply:
done, i see that without default, old configuration treated as invalid and khost starts as new deployment.